### ISSUE TITLE
Cite command cannot handle \ref.

### ISSUE THREAD

**Issue by nickw2066** [zepinglee/citeproc-lua#74]
**Describe the bug**
Cite command cannot handle \ref.

**Additional information**
- TeX distribution: MacTeX 2024
- Package `citation-style-language` v0.6.0
- LaTeX engine: ``lualatex`

**To Reproduce**
```tex
\documentclass{article}

\begin{filecontents}[overwrite, noheader]{\jobname.json}
[
    {
        "id": "King2008-Gramma",
        "type": "thesis",
        "author": [
            {
                "family": "King",
                "given": "John T."
            }
        ],
        "title": "A Grammar of Dhimal",
        "title-short": "Grammar of Dhimal",
        "source": "PDF",
        "publisher-place": "Netherlands",
        "publisher": "Universiteit Leiden",
        "URL": "https://scholarlypublications.universiteitleiden.nl/handle/1887/13072",
        "genre": "doctoral dissertation",
        "issued": {
            "date-parts": [
                [
                    "2008",
                    "9",
                    "3"
                ]
            ]
        },
        "accessed": {
            "date-parts": [
                [
                    "2024",
                    "08",
                    "13"
                ]
            ]
        }
    }
]
\end{filecontents}

\usepackage{tikz}

\usepackage{citation-style-language}
\cslsetup{style = turabian-fullnote-bibliography}
\addbibresource{\jobname.json}

\begin{document}

See Figure \ref{fig:example}.\footnote{Footnote reference to
Figure \ref{fig:example} is fine.}

\begin{figure}[h]
    \centering
    \begin{tikzpicture}
        \draw[thick] (0,0) rectangle (4,2);
    \end{tikzpicture}
    \caption{This is a simple example figure.}
    \label{fig:example}
\end{figure}

Notice that \texttt{cite} command with a reference
to Figure \ref{fig:example} in the suffix cannot render the figure
number.\cite[suffix = {; see Figure \ref{ig:example}.}]{King2008-Gramma}

\end{document}
```

**Screenshots**
![Screenshot 2024-08-21 at 3 31 21 PM](https://github.com/user-attachments/assets/9f836322-0cb4-448c-8469-528496d8be93)

CSL: [https://www.zotero.org/styles/turabian-fullnote-bibliography](https://www.zotero.org/styles/turabian-fullnote-bibliography)

**Comment by zepinglee** [comment]
It's because of the missing `f` in `\ref{ig:example}`. It works correctly with `\ref{fig:example}`.

<img width="654" alt="Screenshot 2024-08-21 at 17 15 46" src="https://github.com/user-attachments/assets/8ca22f86-fe75-485e-9a2a-0a60a092ecff">

**Comment by nickw2066** [comment]
Very sorry to waste your time. Here is the corrected file:
```tex
\documentclass{article}

\begin{filecontents}[overwrite, noheader]{\jobname.json}
[
    {
        "id": "King2008-Gramma",
        "type": "thesis",
        "author": [
            {
                "family": "King",
                "given": "John T."
            }
        ],
        "title": "A Grammar of Dhimal",
        "title-short": "Grammar of Dhimal",
        "source": "PDF",
        "publisher-place": "Netherlands",
        "publisher": "Universiteit Leiden",
        "URL": "https://scholarlypublications.universiteitleiden.nl/handle/1887/13072",
        "genre": "doctoral dissertation",
        "issued": {
            "date-parts": [
                [
                    "2008",
                    "9",
                    "3"
                ]
            ]
        },
        "accessed": {
            "date-parts": [
                [
                    "2024",
                    "08",
                    "13"
                ]
            ]
        }
    }
]
\end{filecontents}

\usepackage{tikz}

\usepackage{citation-style-language}
\cslsetup{style = turabian-fullnote-bibliography}
\addbibresource{\jobname.json}

\begin{document}

See Figures \ref{exampleFig} and \ref{example_fig}, identical
figures with different labels.\footnote{Footnote reference to
Figures \ref{exampleFig} and \ref{example_fig} is fine.}

\begin{figure}[ht]
    \centering
    \begin{tikzpicture}
        \draw[thick] (0,0) rectangle (4,2);
    \end{tikzpicture}
    \caption{Example figure 1.}
    \label{exampleFig}
\end{figure}
\begin{figure}[ht]
    \centering
    \begin{tikzpicture}
        \draw[thick] (0,0) rectangle (4,2);
    \end{tikzpicture}
    \caption{Example figure 2.}
    \label{example_fig}
\end{figure}

There is no problem if the \texttt{cite} command has a reference
to Figure \ref{exampleFig} in the
prefix/suffix.\cite[suffix = {; see also Figure \ref{exampleFig}.}]{King2008-Gramma}

However, if the \texttt{cite} command has a reference
to Figure \ref{example_fig} in the prefix/suffix then the \LaTeX\ file does not compile
(``Missing \textbackslash endcsname inserted'').
%\cite[suffix = {; see also Figure \ref{example_fig}.}]{King2008-Gramma}

\end{document}
```

**Comment by zepinglee** [comment]
Thanks for your bug report and I can reproduce that. It's because the underscore is incorrectly escaped to `\_` when outputted.

**Comment by zepinglee** [comment]
Fixed in [7b3c8cc](https://github.com/zepinglee/citeproc-lua/commit/7b3c8cc6dd230470f45a5843ac90261be4011747).
